azurerm/internal/services/eventhub/eventhub_resource.go

```diff
@@ -273,8 +273,8 @@ func resourceArmEventHubDelete(d *schema.ResourceData, meta interface{}) error {
 func ValidateEventHubPartitionCount(v interface{}, _ string) (warnings []string, errors []error) {
 	value := v.(int)
 
-	if !(32 >= value && value >= 1) {
-		errors = append(errors, fmt.Errorf("EventHub Partition Count has to be between 1 and 32"))
+	if !(1024 >= value && value >= 1) {
+		errors = append(errors, fmt.Errorf("EventHub Partition Count has to be between 1 and 32 for standard hubs and 1024 for Event Hubs Clusters"))
```

Comment:
Given that this would break validation for non is there any way we could validate its <32 for non clustered eventhubs?

Reply:
If someone can suggest a way to check whether namespace is allocated on a cluster or not - happy to implement.

Reply:
Problem with checking this is really the abstraction of the relationship. EventHubs are created in an EventHub Namespace, which resides on the cluster. Namespace is blissfully unaware of cluster type it lives on, so is EventHub itself.. Any suggestions how to validate it are more than welcome.

Reply:
This won't be possible to do comprehensively in the schema. The two potential sets of boundaries are incompatible, and it's not possible to determine if it's a dedicated cluster before the schema is validated.  The only option would to check against the min-max of the two together and deal with finer grained checks in code, though this is less than ideal.

This may be moot for the moment as the Namespace resource does not yet support the necessary property to create it on a custom cluster afaik, (assuming I've understood `ClusterArmID` correctly in the model).  It may actually be more appropriate to create new, separate resources for `azurerm_eventhub_namespace_dedicated` and `azurerm_eventhub_dedicated` (or similar), since limits and potential behaviours are sufficiently different.  Thoughts?

Reply:
Oh crack, you're completely right, namespace doesn't support this yet. Haven't thought of that, good catch! Eventhubs themselves are working on a namespace that was created manually on dedicated cluster, so the only difference there will be the limits, but namespace is indeed something that can be done. That said, I might just go with an optional clusterID in the current namespace, rather than creating a whole new resource?

Reply:
Given the challenges of validation already discussed, and I'm reasonably sure there will be others, I think the separate resources my be better suited (since we can't cross-reference other schema items to know which boundaries are appropriate) - This would mean we can pick up errors at plan, rather than having to get part way through an apply and have to throw an error.  

If the service changes significantly (it has been in preview for >2yrs) to bring things closer together they can be combined/deprecated later, but if they diverge further we'll be well placed to absorb any breaking technical / behavioural change that may be introduced.

Reply:
Let's see... taking a stab at dedicated resources, although this would be my first time doing so, so please bear with the newbie while I'm trying to get it to work.

Reply:
Thanks @favoretti 

I think you'll be able to use the existing resources as a good start.

If you're not already a member, there's a contributor Slack you can join where we try to help out as much as we can, as well as other contributors that can assist too, the link is in the [README.md](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/README.md) in the root of the project.